server/backend/check.ml

```diff
@@ -323,7 +323,7 @@ fi |} pkg pkg pkg (Server_configfile.platform_distribution conf)
 
         (* attempt to install depexts *)
         (* sometimes it turns out that there is no solution, then exit early *)
-        {|opam install ./ --depext-only --with-test --with-doc || (echo "opam-health-check: Depext unsolvable" && exit 1)|};
+        {|opam install ./ --depext-only --with-test --with-doc | grep -q "ERROR" && (echo "opam-health-check: Depext unsolvable" && exit 1)|};
```

Comment:
Won't it exit with the error code `1` if you are not able to `grep` the `"ERROR"`? On my machine, if `grep` fails, it exits with `1`.

Reply:
Indeed, potentially a better solution could be to write the stderr (presumably the error is getting written to stderr) into a file, checking its contents and failing in that case.

I wonder, would it make sense to report this behavior upstream to opam? It seems like a bug to me that the error code is `0` in such cases.

Reply:
Good catch! The command was exiting even if there wasn't an error. Modified it to write to a file and then grep for error from the file.

Will open an issue on opam.

Reply:
Good idea with `tee`, that safes an invocation of `cat`. I think the code is missing a `;`, the compiler complains about a string applied to another string.

Are you sure the error message is output on stdout?

Reply:
You're right, the error message is displayed in stderr. I've changed the code to only redirect stderr to the temporary file.